src/io/hdf5/HDF5IO.cpp

```diff
@@ -82,7 +83,305 @@ Status HDF5IO::flush()
   return checkStatus(status);
 }
 
-Status HDF5IO::createAttribute(const BaseDataType& type,
+std::unique_ptr<H5::Attribute> HDF5IO::getAttribute(const std::string& path)
+{
+  // Split the path to get the parent object and the attribute name
+  size_t pos = path.find_last_of('/');
+  if (pos == std::string::npos) {
+    return nullptr;
+  }
+
+  std::string parentPath = path.substr(0, pos);
+  std::string attrName = path.substr(pos + 1);
+
+  try {
```

Comment:
could use `getH5ObjectType` here instead of trying to open object as group or a dataset?